File: src/test/java/com/technews/aggregate/posts/spring/dto/SavePostRequestTest.java
package com.technews.aggregate.posts.spring.dto;

import com.technews.aggregate.posts.dto.SavePostRequest;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

class SavePostRequestTest {

    @Test
    void when_isLatestDatePost_then_return_true() {
        final SavePostRequest release = SavePostRequest.builder()
                .date("2024-04-30")
                .build();

        assertTrue(release.isLatestDatePost("2024-04-29"));
        assertTrue(release.isLatestDatePost("2024-04-26"));
    }

    @Test
    void when_isLatestDatePost_then_return_false() {
        final SavePostRequest release = SavePostRequest.builder()
                .date("2024-04-30")
                .build();

        assertFalse(release.isLatestDatePost("2024-04-30"));
        assertFalse(release.isLatestDatePost("2024-04-31"));
    }
}

File: src/test/java/com/technews/aggregate/posts/spring/dto/SavePostRequestTest.kt
package com.technews.aggregate.posts.spring.dto

import com.technews.aggregate.posts.dto.SavePostRequest
import io.kotest.assertions.assertSoftly
import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.booleans.shouldBeFalse
import io.kotest.matchers.booleans.shouldBeTrue

class SavePostRequestTest : StringSpec({
    "latest post should be return true" {
        val release = SavePostRequest(
            date = "2024-04-30",
        )
        assertSoftly(release) {
            release.isLatestDatePost("2024-04-29").shouldBeTrue()
            release.isLatestDatePost("2024-04-26").shouldBeTrue()
        }
    }

    "not latest post should be return false" {
        val release = SavePostRequest(
            date = "2024-04-30",
        )
        assertSoftly(release) {
            release.isLatestDatePost("2024-04-30").shouldBeFalse()
            release.isLatestDatePost("2024-04-31").shouldBeFalse()
        }
    }
})
